File: Projekti/SQLAlchemy-Benchmark/util/stopwatch.py
import time


class Stopwatch:
    def __init__(self):
        self.start_time = None
        self.end_time = None

    def start(self):
        self.start_time = time.time()

    def stop(self):
        self.end_time = time.time()

    def get_time(self):
        if self.start_time is None or self.end_time is None:
            raise Exception("Must call start() and stop() before get_time()")
        return self.end_time - self.start_time

    def get_milliseconds(self):
        if self.start_time is None or self.end_time is None:
            raise Exception("Must call start() and stop() before get_time()")
        return round((self.end_time - self.start_time)*1000)

File: Projekti/JDBC-Benchmark/src/main/java/com/example/JDBCBenchmark/util/Stopwatch.java
package com.example.HibernateBenchmark.util;

public class Stopwatch {
    private long startTime = 0;
    private long endTime = 0;

    public void start() {
        this.startTime = System.currentTimeMillis();
    }

    public void stop() {
        this.endTime = System.currentTimeMillis();
    }

    public long getElapsedMilliseconds() {
        return endTime - startTime;
    }

    public double getElapsedSeconds() {
        return ((double)(endTime - startTime)) / 1000;
    }
}
